### PR TITLE
[WIN32K] Fix alignment checks in NtUserSetInformationThread

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose

Fixes random failures on x64.

### CODE REVIEW

**Review by JoachimHenze — APPROVED** [review]
I don't understand it and have the same question in mind as Hermes. Would love to hear the explanation. But this doesn't bloat anything, is small, and Timo proved many times having a track record for good x64 patches and good intentions. So if my trust-checkmark helps him to get his work done faster, I will happily give it here.

File: win32ss/user/ntuser/ntstubs.c [hunk 1]
```diff
@@ -783,7 +783,7 @@ NtUserSetInformationThread(IN HANDLE ThreadHandle,
             Status = STATUS_SUCCESS;
             _SEH2_TRY
             {
-                ProbeForRead(ThreadInformation, sizeof(CsrPortHandle), sizeof(PVOID));
+                ProbeForRead(ThreadInformation, sizeof(CsrPortHandle), sizeof(ULONG));
```
[the remark this answers is not in the document]
Comment (tkreuzer) [reply win32ss/user/ntuser/ntstubs.c]: I "fixed" it. If you decide to validate this with a test (you would need to inject code into csrss and then throw a user APC at the csrss thread to even execute it), you might find that this is not actually correct, but it doesn't matter at all.

**Review by katahiromz — APPROVED** [review]

File: win32ss/user/ntuser/ntstubs.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (HBelusca) [reply win32ss/user/ntuser/ntstubs.c]: And what would be a "correct" fix then?

File: win32ss/user/ntuser/ntstubs.c [hunk 2]
```diff
@@ -783,7 +783,7 @@ NtUserSetInformationThread(IN HANDLE ThreadHandle,
             Status = STATUS_SUCCESS;
             _SEH2_TRY
             {
-                ProbeForRead(ThreadInformation, sizeof(CsrPortHandle), sizeof(PVOID));
+                ProbeForRead(ThreadInformation, sizeof(CsrPortHandle), _alignof(CsrPortHandle));
```
Comment (HBelusca) [inline win32ss/user/ntuser/ntstubs.c:786]: single-underscore alignof?
Note that we also have:
[#define TYPE_ALIGNMENT(t) __alignof(t)](https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/include/xdk/ntbasedef.h;hb=be8a6f8af19b7b9f7be29ea6b81ed8da029daa2c#l113)
acts on a type though.

Comment (tkreuzer) [reply win32ss/user/ntuser/ntstubs.c]: Fixed using the double underscore version.

**Comment by tkreuzer** [comment]
> I don't understand it and have the same question in mind as Hermes. 

There is no explanation other than "what Windows does" and it doesn't matter here, because both works and it isn't exposed to apps. So either is fine.

File: win32ss/user/ntuser/ntstubs.c [hunk 3]
```diff
@@ -783,7 +783,7 @@ NtUserSetInformationThread(IN HANDLE ThreadHandle,
             Status = STATUS_SUCCESS;
             _SEH2_TRY
             {
-                ProbeForRead(ThreadInformation, sizeof(CsrPortHandle), sizeof(PVOID));
+                ProbeForRead(ThreadInformation, sizeof(CsrPortHandle), __alignof(CsrPortHandle));
```
Comment (HBelusca) [inline win32ss/user/ntuser/ntstubs.c:786]: I was mentioned in a previous comment that `__alignof` acts on a type according to MSDN
https://learn.microsoft.com/en-us/cpp/cpp/alignof-operator?view=msvc-170
but you make it act on a variable. Is this correctly supported and does it behave correctly?

**Pull request merged**